packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -51,17 +50,59 @@ module O = {
     | None => rFn()
     }
   ()
-  let fmap = Rationale.Option.fmap
-  let bind = Rationale.Option.bind
-  let default = Rationale.Option.default
-  let isSome = Rationale.Option.isSome
-  let isNone = Rationale.Option.isNone
-  let toExn = Rationale.Option.toExn
-  let some = Rationale.Option.some
-  let firstSome = Rationale.Option.firstSome
-  let toExt = Rationale.Option.toExn // wanna flag this-- looks like a typo but `Rationale.OptiontoExt` doesn't exist.
-  let flatApply = (fn, b) => Rationale.Option.apply(fn, Some(b)) |> Rationale.Option.flatten
-  let flatten = Rationale.Option.flatten
+  let fmap = (f: 'a => 'b, x: option<'a>): option<'b> => {
+    switch x {
+    | None => None
+    | Some(x') => Some(f(x'))
+    }
+  }
+  let bind = (o, f) =>
+    switch o {
+    | None => None
+    | Some(a) => f(a)
+    }
+  let default = (d, o) =>
+    switch o {
+    | None => d
+    | Some(a) => a
+    }
+  let isSome = o =>
+    switch o {
+    | Some(_) => true
+    | _ => false
+    }
+  let isNone = o =>
+    switch o {
+    | None => true
+    | _ => false
+    }
+  let toExn = (err, o) =>
+    switch o {
+    | None => raise(Failure(err))
+    | Some(a) => a
+    }
+
+  let some = a => Some(a)
+  let firstSome = (a, b) =>
+    switch a {
+    | None => b
+    | _ => a
+    }
+
```

Comment:
There are newlines between some functions but not others

Reply:
Weird. I just didn't pay attention to `rescript format`, but I can take a pass through.